Add avatar of message senders to conversation view

ben181231 opened the pull request:
connect #56

carmenlau commented:
Only one minor comment, when user's name is null. The app will crash as we cannot generate avatar for him. I think it would be better to have empty avatar instead of crash.

ben181231 commented:
@carmenlau 

Oh. I know that. 

As discussed before, we would try to use `username` of the user record instead of `name` field which requires an extra update during sign up. However, the field ACL for `username` field is not readable for public, that is why I moved back to use `name` field in 1b8142f.

Should we changed to `Unknown` if the user is not set `user` field and give a background-only avatar to him / her ?

carmenlau commented:
We will follow iOS implementation, using `name` attribute in user record for name display. If it is empty we will use `Unknown` for user name and background-only avatar

carmenlau commented on chat/src/main/java/io/skygear/plugins/chat/ui/model/User.kt:
```diff
@@ -34,16 +40,26 @@ class User : IUser {
 
     override fun getAvatar(): String {
-        return this.chatUser.record.get(User.AvatarField) as String
+        val avatarUrl = this.chatUser.record.get(User.AvatarField) as String?
```
As I remember the AvatarField should be assets instead of string? I think we should write down the assumption like user profile need to have `name` as string and the avatar fields. Besides that, the PR looks fine. Let me merge it first and I will open issue to see if we need to change it to assets :)
